extensions/aria_extension_tosca/simple_v1_0/modeling/__init__.py

```diff
@@ -570,22 +578,23 @@ def added_all():
                         container.children.append(model)
 
 
-def create_parameter_models_from_values(properties, source_properties):
+def create_parameter_models_from_values(properties, source_properties, model_class=None):
+
     if source_properties:
         for property_name, prop in source_properties.iteritems():
-            properties[property_name] = Parameter(name=property_name, # pylint: disable=unexpected-keyword-arg
-                                                  type_name=prop.type,
-                                                  value=prop.value,
-                                                  description=prop.description)
+            properties[property_name] = model_class(name=property_name, # pylint: disable=unexpected-keyword-arg
+                                                    type_name=prop.type,
+                                                    value=prop.value,
+                                                    description=prop.description)
 
 
 def create_parameter_models_from_assignments(properties, source_properties):
```

Comment:
Decide if this is a general function for Parameters, or a specific function for Properties.